File: frontend/src/types/governance.ts
export interface BOSV9RCManifestSafetyBoundary {
  release_decision: "review_required";
  model_version: "ready_for_review";
  final_action_draft_count: 0;
  final_action_audit_record_count: 0;
  external_share_record_created: false;
  final_action_execute_call_count: 0;
  validated_default_write_enabled: false;
  runtime_activation_enabled: false;
  hardware_execution_enabled: false;
  final_action_execution_enabled: false;
}

export interface BOSV9RCManifestCheck {
  status: string;
  raw_status?: string;
  name: string;
  summary: string;
}

export interface BOSV9RCManifestScorecardItem {
  dimension: string;
  score: number;
  target: number;
  evidence: string;
}

export interface BOSV9RCManifestResponse {
  generated_at: string;
  release_state: "review_required";
  candidate_evidence_state: "pending" | "review_required" | "ready_for_review";
  preflight_pass_count: number;
  preflight_warn_count: number;
  preflight_fail_count: number;
  review_required_items: string[];
  local_validation_notes: string[];
  recommended_next_action: string | null;
  bos_v9_target_progress_percent: number;
  bos_v9_target_scorecard: BOSV9RCManifestScorecardItem[];
  safety_boundary: BOSV9RCManifestSafetyBoundary;
  release_reference_smoke_passed: boolean;
  checks: BOSV9RCManifestCheck[];
  artifacts: Record<string, string>;
}

File: backend/app/schemas/governance.py
"""Shared governance envelope schemas for release-facing BOS surfaces."""

from __future__ import annotations

from typing import Any, Literal

from pydantic import BaseModel, ConfigDict, Field


GateState = Literal["review_required"]
ActionBoundary = Literal["forbidden", "requires_confirmation"]
ClaimStatus = Literal["validated", "supported_not_closed", "planned"]


class ReleaseGovernanceActionBoundary(BaseModel):
    action: str
    boundary: ActionBoundary
    reason: str


class ReleaseGovernanceClaim(BaseModel):
    claim: str
    status: ClaimStatus
    source_boundary: str


class ReleaseGovernanceEnvelope(BaseModel):
    schema_version: str = "release_governance_envelope_v1"
    gate_state: GateState = "review_required"
    evidence_chain_id: str | None = None
    source_boundary: str
    human_review_required: bool = True
    review_required_reason: str
    validated_default_write_enabled: bool = False
    final_action_execution_enabled: bool = False
    runtime_activation_enabled: bool = False
    hardware_execution_enabled: bool = False
    external_share_status: ActionBoundary = "requires_confirmation"
    action_boundaries: list[ReleaseGovernanceActionBoundary] = Field(default_factory=list)
    claim_ledger: list[ReleaseGovernanceClaim] = Field(default_factory=list)
    guardrails: list[str] = Field(default_factory=list)


class BOSV9RCManifestSafetyBoundary(BaseModel):
    release_decision: Literal["review_required"]
    model_version: Literal["ready_for_review"]
    final_action_draft_count: Literal[0]
    final_action_audit_record_count: Literal[0]
    external_share_record_created: Literal[False]
    final_action_execute_call_count: Literal[0]
    validated_default_write_enabled: Literal[False]
    runtime_activation_enabled: Literal[False]
    hardware_execution_enabled: Literal[False]
    final_action_execution_enabled: Literal[False]


class BOSV9RCManifestResponse(BaseModel):
    model_config = ConfigDict(extra="allow")

    generated_at: str
    release_state: Literal["review_required"]
    candidate_evidence_state: Literal["pending", "review_required", "ready_for_review"]
    preflight_pass_count: int
    preflight_warn_count: int
    preflight_fail_count: int
    review_required_items: list[str]
    local_validation_notes: list[str]
    recommended_next_action: str | None = None
    bos_v9_target_progress_percent: int
    bos_v9_target_scorecard: list[dict[str, Any]]
    safety_boundary: BOSV9RCManifestSafetyBoundary
    release_reference_smoke_passed: bool
    checks: list[dict[str, Any]]
    artifacts: dict[str, str]
